cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/search/AzureSearch.scala

```diff
@@ -273,6 +304,26 @@ object AzureSearchWriter extends IndexParser with SLogging {
         UDFUtils.oldUdf(checkForErrors(fatalErrors) _, ErrorUtils.ErrorSchema)(col("error"), col("input")))
   }
 
+  private def getVectorColNameTypeTupleFromIndexJson(indexJson: String): Seq[(String, String)] = {
+    parseIndexJson(indexJson).fields
+      .filter(f => f.vectorSearchConfiguration.nonEmpty && f.dimensions.nonEmpty)
+      .map(f => (f.name, f.`type`))
+  }
+  private def castDFColsToVectorCompatibleType(vectorColNameTypeTuple: Seq[(String, String)],
+                                               df: DataFrame): DataFrame = {
+
+    vectorColNameTypeTuple.foldLeft(df) { case (accDF, (colName, colType)) =>
+      assert(accDF.columns.contains(colName), s"Column $colName not found in dataframe columns ${accDF.columns.toList}")
```

Comment:
Just noticed you can also skip the field if you don't want to send it to ACS, and in the case of vectors it just keeps an empty vector. So this check might be unnecessary. I will remove the check, and replace it with a filter on whether the column exists in the schema. We might still give a warning to the user that the column doesn't exist.